src/angular/components/sso.component.ts

```diff
@@ -109,10 +109,14 @@ export class SsoComponent {
             if (returnUri) {
                 state += `_returnUri='${returnUri}'`;
             }
-
-            await this.storageService.save(ConstantsService.ssoStateKey, state);
         }
 
+        // Add Organization Identifier to state
+        state += `_identifier=${this.identifier}`;
+
+        // Save state (regardless of new or existing)
+        await this.storageService.save(ConstantsService.ssoStateKey, state);
```

Comment:
State can be passed in via query string, which I think is why it existed behind the null check. Is there still a need to save it in that case?

Reply:
That is correct - but this is the only place where we're adding on the `orgIdentifier`, so an update to whats in storage is necessary.